lib/image_composition.dart

```diff
@@ -0,0 +1,158 @@
+import 'dart:async';
+import 'dart:ui';
+
+import 'extensions.dart';
+
+export 'extensions.dart';
+
+class _Composed {
+  /// The image that will be composed.
+  final Image image;
+
+  /// The position where the [image] will be composed.
+  final Vector2 position;
+
+  /// The source on the [image] that will be composed.
+  final Rect source;
+
+  /// The angle (in radians) used to rotate the [image] around it's [anchor].
+  final double angle;
+
+  /// The point around which the [image] will be rotated
+  /// (defaults to the centre of the [source]).
+  final Vector2 anchor;
+
+  final bool isAntiAlias;
+
+  /// The [BlendMode] that will be used when composing the [image].
+  final BlendMode blendMode;
+
+  _Composed(
+    this.image,
+    this.position,
+    this.source,
+    this.angle,
+    this.anchor,
+    this.isAntiAlias,
+    this.blendMode,
+  );
+}
+
+class ImageComposition {
+  /// The values that will be used to compose the image
+  final List<_Composed> _composes = [];
+
+  /// The [defaultBlendMode] can be used to change how each image will be
+  /// blended onto the composition. Defaults to [BlendMode.srcOver].
+  final BlendMode defaultBlendMode;
+
+  /// The [defaultAntiAlias] can be used to if each image will be anti aliased.
+  final bool defaultAntiAlias;
+
+  ImageComposition({
+    this.defaultBlendMode = BlendMode.srcOver,
```

Comment:
@required?

Reply:
:-) `required` is not allowed when using default values, and well I dont see why we should make it a required parameter